pkgs/applications/networking/p2p/stig/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, fetchFromGitHub
+, python
+}:
+
+with python.pkgs; buildPythonApplication rec {
+  pname = "stig";
+  version = "0.10.1a";
+
+  src = fetchFromGitHub {
+    owner = "rndusr";
+    repo = "stig";
+    rev = "v${version}";
+    sha256 = "076rlial6h1nhwdxf1mx5nf2zld5ci43cadj9wf8xms7zn8s6c8v";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace "urwidtrees>=1.0.3dev0" "urwidtrees"
```

Comment:
Well actually I haven't had the chance yet to ask the developer about it but I've tested the program with this version of urwidtrees and it seems to be working. I've asked @rnduser upstream as for the matter and I hope he'll be responsive. 

@timokau In case a certain bug will arise because of this different dependency version, how would it be possible to overcome that within Nixpkgs' Python infrastructure?

Reply:
@rnduser said it should be Ok to use version 1.0.2 of urwidtrees, as the need to switch to it was due to an [issue](https://github.com/pazz/urwidtrees/issues/35) which is resolved in 1.0.2 as well.

I'm marking this one as resolved while I've added an appropriate comment near the `postPatch` attribute.

Reply:
Great, thanks for clarifying with upstream. Please link that issue in a code comment so future readers know the justification.

> In case a certain bug will arise because of this different dependency version, how would it be possible to overcome that within Nixpkgs' Python infrastructure?

Unfortunately that's not easy currently. Also see [some of this discussion](https://discourse.nixos.org/t/scaling-the-python-package-set-in-nixpkgs/3749). The options are basically
- update the nixpkgs dependency
- patch the package or otherwise work around the bugs / clearly communicate to the user that some of the functionality is not expected to work.

Reply:
Oh, I see you did already add a comment. Good work :)